src/main/scala/org/zalando/nakadi/client/Connection.scala

```diff
@@ -0,0 +1,185 @@
+package org.zalando.nakadi.client
+
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl._
+import scala.concurrent.Future
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri.apply
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.slf4j.LoggerFactory
+import com.typesafe.scalalogging.Logger
+import org.slf4j.LoggerFactory
+import akka.http.scaladsl.HttpsConnectionContext
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.model.MediaTypes.`application/json`
+import javax.net.ssl.SSLContext
+import java.security.SecureRandom
+import java.security.cert.X509Certificate
+import javax.net.ssl.{ SSLContext, TrustManager, X509TrustManager }
+import scala.concurrent.duration.DurationInt
+import akka.actor.Terminated
+
+trait Client {
+
+  /**
+   * Returns monitoring metrics
+   */
+  def metrics(): Future[HttpResponse]
+
+  /**
+   * Returns a list of all registered EventTypes.
+   */
+  def eventTypes(): Future[HttpResponse]
+
+  /**
+   * Creates a new EventType.
+   *
+   * @param event - The event to create.
+   *
+   */
+  def newEventType(eventType: Any): Future[Boolean]
+
+  /**
+   * Returns the EventType identified by its name.
+   * @param name - Name of the EventType
+   */
+  def eventType(name: String): Future[HttpResponse]
+  /**
+   * Updates the EventType identified by its name.
+   *
+   *
+   * @param name - Name of the EventType
+   * @param event - Event to update
+   */
+  def updateEventType(name: String, event: Any): Future[HttpResponse]
```

Comment:
I would explicitly  call the method createEventType(). IMO the noun based method names are a good fit for retrieval methods (getter, etc) but they are rather confusing for write operations (at least for me)